banks-client/src/lib.rs

```diff
@@ -128,6 +128,17 @@ impl BanksClient {
             .map_err(Into::into)
     }
 
+    pub fn process_transaction_with_preflight_and_context(
+        &mut self,
+        ctx: Context,
+        transaction: Transaction,
+        commitment: CommitmentLevel,
+    ) -> impl Future<Output = Result<BanksTransactionResult, BanksClientError>> + '_ {
```

Comment:
Nice, this way #22058 can come in afterwards, but I'll aim to land it first either way